### PR TITLE
Convert Tail to MSW

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Continued effort to convert tests use MSW

task #1874

### CODE REVIEW

**Review by rozenmd — APPROVED** [review]

File: packages/wrangler/src/__tests__/tail.test.ts [hunk 1]
```diff
@@ -28,7 +27,6 @@ describe("tail", () => {
 	afterEach(() => {
 		mockWebSockets.forEach((ws) => ws.close());
 		mockWebSockets.splice(0);
-		unsetAllMocks();
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/tail.test.ts:31]: Should we not still unset mocks after each run? Maybe I'm misunderstanding what this does

File: packages/wrangler/src/__tests__/tail.test.ts [hunk 2]
```diff
@@ -67,38 +65,101 @@ describe("tail", () => {
...
+							result: [
+								{
+									pattern: "example.com/*",
+									script: "test-worker",
+								},
+							],
+						})
+					);
+				})
+			);
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/tail.test.ts:106]: I think this would be slightly cleaner if it was a reimplementation of `mockGetZoneFromHostRequest` and `mockCollectKnownRoutesRequest`, since this adds a lot of boilerplate to each test

File: packages/wrangler/src/__tests__/tail.test.ts [hunk 3]
```diff
@@ -712,6 +802,15 @@ function mockWebsocketAPIs(
...
 		},
+		/**
+		 * Close the mock websocket and clean up the API.
+		 * The setTimeout forces a cycle to allow for closing and cleanup
+		 * @returns a Promise that resolves when the websocket is closed
+		 */
+		async closeHelper() {
+			api.ws.close();
+			await new Promise((resolve) => setTimeout(resolve, 0));
+		},
```
Comment (penalosa) [inline packages/wrangler/src/__tests__/tail.test.ts:813]: It looks like this was working without the `setTimeout` before—why is this needed now? Would `process.nextTick` work?

File: packages/wrangler/src/__tests__/tail.test.ts (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/tail.test.ts:106]: I will do that with a future PR when other files using those abstractions/helpers are also converted so there is no conflict with the custom mocks and MSW; they conflict when used in the same test file.

File: packages/wrangler/src/__tests__/tail.test.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/tail.test.ts:31]: That unsets the Custom Mocks, MSW has its own cleanup that occurs in Jest setup AND I use the `res.once` which also self cleans up (since we do so much inlining MSW server & handlers in each test)
